### PR TITLE
KNOX-2258 - Add filter for Location header

### PR DESCRIPTION
**Pull request by abhi2196**
## What changes were proposed in this pull request?

Follow up for #276 
Added a new filter in Livy rewrite.xml for replacing Location header
Removed rewrite rules from routes path in service.xml

## How was this patch tested?

Manually tested in a Cluster with Knox and Livy Services.

1. Edited the rewrite/service.xml for Livy service with the changes.
2. Verified that `https://<KNOX_HOST:KNOX_PORT>/gateway/default/livy` gets redirected to `https://<KNOX_HOST:KNOX_PORT>/gateway/default/livy/ui`
3. Verified that `https://<KNOX_HOST:KNOX_PORT>/gateway/default/livy/` gets redirected to `https://<KNOX_HOST:KNOX_PORT>/gateway/default/livy/ui`

### CODE REVIEW

File: gateway-service-definitions/src/main/resources/services/livy/0.4.0/rewrite.xml [hunk 1]
```diff
@@ -49,4 +49,10 @@
     <rewrite template="{$frontend[url]}/livy/ui/"/>
   </rule>
 
+  <filter name="LIVYSERVER/livy/outbound/headers">
```
Comment (moresandeep) [inline gateway-service-definitions/src/main/resources/services/livy/0.4.0/rewrite.xml]: Where is this filter getting called from ? we need a corresponding entry in service.xml
```xml
    <route path="/livy">
        <rewrite apply="LIVYSERVER/livy/outbound/headers" to="response.headers"/>
    </route>
```

File: gateway-service-definitions/src/main/resources/services/livy/0.4.0/service.xml [hunk 2]
```diff
@@ -24,12 +24,8 @@
     </metadata>
   <routes>
     <route path="/livy/**?**"/>
-    <route path="/livy">
```
Comment (moresandeep) [inline gateway-service-definitions/src/main/resources/services/livy/0.4.0/service.xml]: Like mentioned previously we need an entry to call the filter.

File: gateway-service-definitions/src/main/resources/services/livy/0.4.0/rewrite.xml (hunk 1 above)
Comment (abhi2196) [reply gateway-service-definitions/src/main/resources/services/livy/0.4.0/rewrite.xml]: Sure Added the relevant change.
But not sure why it is working in my test cluster without calling this filter explicitly in route path,  are these filters/rules applied by default based on matched pattern ?

File: gateway-service-definitions/src/main/resources/services/livy/0.4.0/service.xml (hunk 2 above)
Comment (abhi2196) [reply gateway-service-definitions/src/main/resources/services/livy/0.4.0/service.xml]: Sure Added the relevant change.

File: gateway-service-definitions/src/main/resources/services/livy/0.4.0/rewrite.xml (hunk 1 above)
Comment (moresandeep) [reply gateway-service-definitions/src/main/resources/services/livy/0.4.0/rewrite.xml]: They should not be, that is weird.

**Pull request merged**